cogs/monitor.py

```diff
@@ -13,96 +23,238 @@ async def on_message(self, message: discord.Message):
+                    await log_suspicious_message(guild_data.monitor_message_log_id, message)
+                    return
+
+    cog_check = cogify(staff_command())
 
     @commands.hybrid_group()
     async def monitor(self, ctx: commands.Context):
         """
         Commands for monitoring problematic message patterns and users.
         """
-        raise NotImplementedError('Command requires implementation and permission set-up.')
+        await available_subcommands(ctx)
 
     @monitor.group()
+    @admin_command()
     async def channel(self, ctx: commands.Context):
         """
         Commands for setting up channels to log problematic messages to.
         """
-        raise NotImplementedError('Command requires implementation and permission set-up.')
+        await available_subcommands(ctx)
 
-    @channel.command()
-    async def user(self, ctx: commands.Context):
+    @channel.command(name='user')
+    @admin_command()
+    async def channel_user(self, ctx: commands.Context, channel: Optional[discord.TextChannel] = None):
         """
         Set logging channel for monitored users.
         """
-        raise NotImplementedError('Command requires implementation and permission set-up.')
+        values = dict(monitor_user_log_id = channel.id if channel else None)
+        await GuildData.update_or_create(values, guild_id=ctx.guild.id)
+        get_guild_data.cache_clear()
+
+        if channel:
+            await ctx.send(f'Log channel for monitored users set to <#{channel.id}> for this guild.')
+        else:
+            await ctx.send('Log channel for monitored users removed for this guild.')
     
-    @channel.command()
-    async def message(self, ctx: commands.Context):
+    @channel.command(name='message')
+    @admin_command()
+    async def channel_message(self, ctx: commands.Context, channel: Optional[discord.TextChannel] = None):
         """
         Set logging channel for monitored message patterns.
         """
-        raise NotImplementedError('Command requires implementation and permission set-up.')
+        values = dict(monitor_message_log_id = channel.id if channel else None)
+        await GuildData.update_or_create(values, guild_id=ctx.guild.id)
+        get_guild_data.cache_clear()
 
+        if channel:
+            await ctx.send(f'Log channel for monitored messages set to <#{channel.id}> for this guild.')
+        else:
+            await ctx.send('Log channel for monitored messages removed for this guild.')
 
     @monitor.command()
     async def cleanup(self, ctx: commands.Context):
         """
         Remove users that are no longer in the server.
         """
-        raise NotImplementedError('Command requires implementation and permission set-up.')
+        monitor_users = await get_all_monitor_users()
+        users_removed = 0
+
+        for monitor_user in monitor_users:
+            if not ctx.guild.get_member(monitor_user.user_id):
```

Comment:
I'm not sure if its possible, but wouldn't it be better to check if the users had been banned as opposed to just not in the server anymore? I believe what is written is currently how its done, so I'm happy with it, but it may be better to change it. What do you think?

Reply:
That makes more sense to me yeah - should be added